libftl/ftl-sdk.c

```diff
@@ -75,14 +79,10 @@ FTL_API ftl_status_t ftl_ingest_create(ftl_handle_t *ftl_handle, ftl_ingest_para
     ftl->status_q.count = 0;
     ftl->status_q.head = NULL;
 
-    os_init_mutex(&ftl->status_q.mutex);
-
     if (os_semaphore_create(&ftl->status_q.sem, "/StatusQueue", O_CREAT, 0) < 0) {
       return FTL_MALLOC_FAILURE;
     }
 
-    os_init_mutex(&ftl->state_mutex);
```

Comment:
If you pass in a completely invalid stream key , like "wrongkey", you will fail on line 51, _get_chan_id_and_key. Hence when you ingest_destroy, state_mutex is not initialized, and deleting it causes error on line 310. 
So this needs to be initialized earlier.